.github/workflows/tests.yml

```diff
@@ -0,0 +1,36 @@
+name: Tests
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+concurrency:
+  group: ${{ github.repository }}-${{ github.ref }}-${{ github.head_ref }}-${{ github.workflow }}
+  cancel-in-progress: true
+jobs:
+  tests_linux:
+    runs-on: ${{ matrix.os }}
+    needs: []
+    timeout-minutes: 60
+    strategy:
+      matrix:
+        python-version: [3.7, 3.8, 3.9, "3.10"]
+        os: ["ubuntu-latest", "macOS-latest", "windows-2019"]
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install Entanglement Forging
+        run: |
+            set -e
```

Comment:
Do you know if `set -e` has any effect here?  I believe that even without this, the Action would stop executing as soon as it had a failing line.  And without this line, you wouldn't need the `shell: bash` line either.

Reply:
Sounds good. i wasnt sure if it would error out or not.

 What does the set command have to do with the shell you use?

Reply:
`set` [is a bash builtin](https://www.gnu.org/software/bash/manual/html_node/The-Set-Builtin.html).  There _might_ be an equivalent in some other shells, but I wouldn't rely on it, particularly for the Windows tests.  We can get by without using anything bash-specific, so no need to invoke bash.